src/Native/Bootstrap/main.cpp

```diff
@@ -28,17 +56,31 @@ __declspec(allocate(".modules$Z")) void * __modules_z[] = { nullptr };
 extern "C" void __managedcode_a();
 extern "C" void __managedcode_z();
 
-#else
-// TODO: Once the dotnet compiler is modified to configure the linker to merge sections
-// on OSX / Linux, store these externs in the .modules section and remove work-around
-// in ModuleHeaderNode.cs.
-//extern "C" __attribute((section ("__DATA,.modules$A"))) void * __modules_a[];
-//extern "C" __attribute((section ("__DATA,.modules$Z"))) void * __modules_z[];
-//__attribute((section ("__DATA,.modules$A"))) void * __modules_a[] = { nullptr };
-//__attribute((section ("__DATA,.modules$Z"))) void * __modules_z[] = { nullptr };
+#else // _MSC_VER
+
+#if defined(__APPLE__)
+
+//
+// On platforms with MachO binaries (OSX), we cannot use the above mechanism to register
```

Comment:
You cannot merge sections on OSX manually and control their ordering. So there's no #pragma comment as on Windows, and no linker script support in `ld`. I have basically implemented the C++ compiler's mechanism for initializing global data on all three platforms now.